app/views/reviews/create.js.erb

```diff
@@ -1,3 +1,9 @@
-document.querySelector('#review-form').reset()
-document.querySelector('#review-count').innerHTML = "<%= review_count(@review.movie) %>"
-document.querySelector('#flash-id').innerHTML = "<%= j render partial: 'layouts/flash' %>"
\ No newline at end of file
+<% if !@review.errors.any? %>
+    document.querySelector('#review-form-container').innerHTML = ""
+    document.querySelector('#form-error').innerHTML = ""
+    document.querySelector('#review-stars').innerHTML = "<%= j render partial: 'shared/review_stars', movie: @movie %>"
+    document.querySelector('#reviews').innerHTML = "<%= j render partial: 'shared/reviews', movie: @movie %>"
+    document.querySelector('#flash-id').innerHTML = "<%= j render partial: 'layouts/flash' %>"
+<% else %>
+    document.querySelector('#form-error').innerHTML = "<%= j render 'shared/errors', object: @review %>"
```

Comment:
Here no need for this much of fine granulation. You just re-render the whole form
`document.querySelector('#review_form').innerHTML = "<%= j render 'form', review: @review %>"`

And then implement errors inside the form. If there is an error show the error.

Yours in-memory object `@review` will carry the errors collection `@review.errors` on which you can iterate or use however you want on the form.